seq2seq/dataset/dataset.py

```diff
@@ -5,36 +5,111 @@ class Dataset(object):
+            will be dropped in the sequences.
+
+        Args:
+            path (str): path to the dataset file
+            src_max_len (int): maximum source sequence length
+            tgt_max_len (int): maximum target sequence length
+            src_vocab (Vocabulary): pre-populated Vocabulary object or a path of a file containing words for the source language,
+            default `None`. If a pre-populated Vocabulary object, `src_max_vocab` wouldn't be used.
+            tgt_vocab (Vocabulary): pre-populated Vocabulary object or a path of a file containing words for the target language,
+            default `None`. If a pre-populated Vocabulary object, `tgt_max_vocab` wouldn't be used.
+            src_max_vocab (int): maximum source vocabulary size
+            tgt_max_vocab (int): maximum target vocabulary size
+        """
+        obj = cls(src_max_len, tgt_max_len)
         pairs = utils.prepare_data(path, src_max_len, tgt_max_len)
+        return cls.encode(obj, pairs, src_vocab, tgt_vocab, src_max_vocab, tgt_max_vocab)
 
+    @classmethod
+    def from_list(cls, src_data, tgt_data, src_max_len, tgt_max_len, src_vocab=None, tgt_vocab=None, src_max_vocab=50000,
+                  tgt_max_vocab=50000):
+        """
+        Initialize a dataset from the source and target lists of sequences.
+
+        Note:
+            Source or target sequences that are longer than the respective
+            max length will be filtered.
+            As specified by maximum vocabulary size, source and target
+            vocabularies will be sorted in descending token frequency and cutoff.
+            Tokens that are in the dataset but not retained in the vocabulary
+            will be dropped in the sequences.
+
+        Args:
+            src_data (list): list of source sequences
+            tgt_data (list): list of target sequences
+            src_max_len (int): maximum source sequence length
+            tgt_max_len (int): maximum target sequence length
+            src_vocab (Vocabulary): pre-populated Vocabulary object or a path of a file containing words for the source language,
+            default `None`. If a pre-populated Vocabulary object, `src_max_vocab` wouldn't be used.
+            tgt_vocab (Vocabulary): pre-populated Vocabulary object or a path of a file containing words for the target language,
+            default `None`. If a pre-populated Vocabulary object, `tgt_max_vocab` wouldn't be used.
+            src_max_vocab (int): maximum source vocabulary size
+            tgt_max_vocab (int): maximum target vocabulary size
+        """
+        obj = cls(src_max_len, tgt_max_len)
+        pairs = utils.prepare_data_from_list(src_data, tgt_data, src_max_len, tgt_max_len)
+        return cls.encode(obj, pairs, src_vocab, tgt_vocab, src_max_vocab, tgt_max_vocab)
+
+    def encode(self, pairs, src_vocab=None, tgt_vocab=None, src_max_vocab=50000, tgt_max_vocab=50000):
```

Comment:
Is `encode` a user-facing method?  Remember not to show it in the doc, or just add an underscore prefix.